### DISCUSSION TITLE
add vw_qgep_channel

### DISCUSSION THREAD

**Discussion by cymed** [QGEP/QGEP#794] [Ideas]
Hi everyone

At the moment, qgep creates one channel per reach. Assigning one channel to multiple reaches is rather bothersome (select both reaches and alter to join the channels of a reach). In order to select channels by clicking I'd find it practical to add a view "vw_qgep_channel" onto which we can then build a reference. I have built such a view [(see here)](https://github.com/QGEP/datamodel/compare/master...cymed:datamodel:patch-4) but I struggle with the corresponding view, yaml and .py scripts.

Does anyone else support the idea and could write the corresponding scripts?

**Comment by sjib** [comment]
@cymed There is the feature idea of a split / merge reach tool and some discussion on it already here:

- [QGEP tool for splitting reaches #527](https://github.com/QGEP/QGEP/issues/527)
- [3D merging and cutting reaches #535](https://github.com/QGEP/QGEP/issues/535)
- [Merging features #220 - QGIS has some basic tools for merging](https://github.com/QGEP/QGEP/issues/220)
Can you look thru and identify if your idea is already covered by these thoughts or this is a new idea?
You can also see who is interested in such tasks.

**Comment by cymed** [comment]
I am also interested in the development of #527, but it can only be used to digitise new channels & reaches. 
#535 can also be practical when a pwwf reach joins was split by a pwwf reach that is downgraded to swwf.
However, when importing a dataset that does not distinguish between reach and channel (i.e. VSA-DSS mini), I'd like to be able to merge channels without merging reaches. For this, I'd like to use my new view.

**Comment by cymed** [comment]
Additional input after discussion with @urskaufmann at the QGIS-CH general assembly: The view can also be practical for tv inspections, as the operators are only interested in the channels and not the reaches. The code returns multilinestrings at the moment, but lines are joined at their endpoints at 2-way intersections (which is the normal case for channels anyway).
